src/test/ui/coherence/coherence-fundamental-trait-objects.old.stderr

```diff
@@ -2,9 +2,11 @@ error[E0117]: only traits defined in the current crate can be implemented for ar
   --> $DIR/coherence-fundamental-trait-objects.rs:15:1
    |
 LL | impl Misc for dyn Fundamental<Local> {}
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl doesn't use types inside crate
+   | ^^^^^^^^^^^^^^----------------------
+   | |             |
+   | |             `dyn coherence_fundamental_trait_lib::Fundamental<Local>` is not defined in the current crate
```

Comment:
I think we can get away with just printing the path, without `dyn` or the generics, etc. E.g. here we would just have:
```
`coherence_fundamental_trait_lib::Fundamental` is not defined in the current crate
```
because `dyn` and `<Local>` are red herrings: they're not what's causing the error (if they were, we should have specific error messages calling them out, like array impls).

Reply:
Ah, sorry, I just read your comment here: https://github.com/rust-lang/rust/pull/65318#discussion_r334290164. Sorry if I seem like a broken record: I'm just worried that these more specific error messages might be more confusing when they contain this extra information.

Reply:
I'm looking over the current test output, and the only cases where we mention `dyn` already have `dyn` in the code (or would if the user follows the warnings). The `dyn` is written unconditionally in the pprinting machinery, and removing would only work for some cases (you could have `(dyn T + K)`, which couldn't be represented with acurately bare path).